### PR TITLE
task/#144646 Feature manage list logic P1

### PR DESCRIPTION
**Pull request by daolq3012**
- Call API fill data into list
- Accept/ Reject member requests
- Redmine: https://dev.framgia.com/redmine/issues/144646
- Effect: 3 screen manage request

### CODE REVIEW

File: src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift [hunk 1]
```diff
@@ -27,11 +33,97 @@ class ManageRequestListViewController: BaseViewController {
...
+    
+    func getListManageRequest(page: Int = 1) {
+        if isLoading {
+            return
+        }
+        AlertHelper.showLoading()
+        isLoading = true
+        switch requestType {
+        case .others:
+            ManageRequestProvider.getListManageLeaveRequest(requestType: requestType, manageRequestApiInput: manageRequestApiInput)
```
Comment (duongdang152) [inline src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift]: tách hàm

File: src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift [hunk 2]
```diff
@@ -27,11 +33,97 @@ class ManageRequestListViewController: BaseViewController {
...
+                    self.currentPage = page
+                }.catch { error in
+                    AlertHelper.showError(error: error)
+                }.always {
+                    AlertHelper.hideLoading()
+                    self.refreshControl.endRefreshing()
+                    self.isLoading = false
+            }
+        case .overTime:
+            ManageRequestProvider.getListManageOverTimeRequest(requestType: requestType, manageRequestApiInput: manageRequestApiInput)
```
Comment (duongdang152) [inline src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift]: tách hàm

File: src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift [hunk 3]
```diff
@@ -27,11 +33,97 @@ class ManageRequestListViewController: BaseViewController {
...
+                    self.currentPage = page
+                }.catch { error in
+                    AlertHelper.showError(error: error)
+                }.always {
+                    AlertHelper.hideLoading()
+                    self.refreshControl.endRefreshing()
+                    self.isLoading = false
+            }
+        case .dayOff:
+            ManageRequestProvider.getListManageOffRequest(requestType: requestType, manageRequestApiInput: manageRequestApiInput)
```
Comment (duongdang152) [inline src/wsm/wsm/viewcontrollers/ManageRequestListViewController.swift]: tách hàm

**Comment by daolq3012** [comment]
OK!

**Pull request merged**